examples/events/basicevent/basicevent.go

```diff
@@ -0,0 +1,121 @@
+package basicevent
+
+import (
+	"errors"
+	"github.com/golang/glog"
+	"time"
+
+	"github.com/mishudark/triper"
+)
+
+
+var (
+	//ErrSubjectMissing When an event has no subject id
+	ErrSubjectMissing = errors.New("SubjectId not found")
+	//ErrCreatedAtMissing When an event has no cratedAt date
+	ErrCreatedAtMissing = errors.New("CreatedAt not found")
+)
+
+//BasicEvent is the base structure for the events
+type BasicEvent struct {
+	triper.BaseAggregate
+	SubjectId string
+	Payload   map[string]string
+	Source    string
+	Tag       string
+	Target    string
+	CreatedAt time.Time
+}
+
+//Assign the values to the corresponding fields on an event
+func (event *BasicEventCreated) CleanEvent() (*BasicEventCreated, error) {
+
+	if val, ok := event.Payload["subject_id"]; ok {
+		event.SubjectId = val
+		delete(event.Payload, "subject_id")
+	} else {
+		return nil, ErrSubjectMissing
+	}
+
+	if val, ok := event.Payload["created_at"]; ok {
+		t, err := time.Parse(time.RFC3339, val)
+		if err != nil {
+			glog.Fatalln(err)
+		}
+		event.CreatedAt = t
+		delete(event.Payload, "created_at")
+	} else {
+		return nil, ErrCreatedAtMissing
+	}
+
+	if val, ok := event.Payload["source"]; ok {
+		event.Source = val
+		delete(event.Payload, "source")
+	}
+
+	if val, ok := event.Payload["target"]; ok {
+		event.Target = val
+		delete(event.Payload, "target")
+	}
+
+	return event, nil
+}
+
+//ApplyChange to the Event
```

Comment:
comment should have the same name as the exported method, `// Reduce`

Reply:
👍